File: example/systems/ParallaxStarRepeatingSystem.java
package net.brokenspork.systems;

import net.brokenspork.components.ParallaxStar;
import net.brokenspork.components.Position;
import net.brokenspork.core.Constants;

import com.artemis.Aspect;
import com.artemis.ComponentMapper;
import com.artemis.Entity;
import com.artemis.annotations.Mapper;
import com.artemis.systems.IntervalEntityProcessingSystem;

public class ParallaxStarRepeatingSystem extends IntervalEntityProcessingSystem {
	@Mapper
	ComponentMapper<Position> pm;

	@SuppressWarnings("unchecked")
    public ParallaxStarRepeatingSystem() {
		super(Aspect.getAspectForAll(ParallaxStar.class, Position.class), 1);
	}

	@Override
	protected void process(Entity e) {
		Position position = pm.get(e);

		if (position.y < -Constants.FRAME_HEIGHT / 2) {
			position.y = Constants.FRAME_HEIGHT / 2;
		}
	}

}

File: example/systems/ParallaxStarRepeatingSystem.ts
module brokenspork.systems {
	
	import ParallaxStar = brokenspork.components.ParallaxStar;
	import Position = brokenspork.components.Position;
	import Constants = brokenspork.core.Constants;
	
	import Aspect = artemis.Aspect;
	import ComponentMapper = artemis.ComponentMapper;
	import Entity = artemis.Entity;
	import IntervalEntityProcessingSystem = artemis.systems.IntervalEntityProcessingSystem;
	import Mapper = artemis.annotations.Mapper;
	
	export class ParallaxStarRepeatingSystem extends IntervalEntityProcessingSystem {
		@Mapper(Position) pm:ComponentMapper<Position>;
	
		//@SuppressWarnings("unchecked")
		constructor() {
			super(Aspect.getAspectForAll(ParallaxStar, Position), 1);
		}
	
		
		public processEach(e:Entity) {
			var position:Position = this.pm.get(e);
	
			if (position.y < -Constants.FRAME_HEIGHT / 2) {
				position.y = Constants.FRAME_HEIGHT / 2;
			}
		}
	
	}
}

